code/core/src/core-server/dev-server.ts

```diff
@@ -130,5 +132,28 @@ export async function storybookDevServer(options: Options) {
   // Now the preview has successfully started, we can count this as a 'dev' event.
   doTelemetry(app, core, initializedStoryIndexGenerator, options);
 
+  // Duplicative to withTelemetry's cancelTelemetry because
+  // - this is a long-running process and withTelemetry has already exited
+  // - we want to report the index stats and don't have it e.g. init
+  async function cancelTelemetry() {
+    const payload = { eventType: 'dev' };
+    try {
+      const generator = await initializedStoryIndexGenerator;
+      const indexAndStats = await generator?.getIndexAndStats();
+      if (indexAndStats) {
+        Object.assign(payload, {
+          storyIndex: summarizeIndex(indexAndStats.storyIndex),
+          storyStats: indexAndStats.stats,
+        });
+      }
+    } catch (err) {}
+    await telemetry('canceled', payload, { immediate: true });
```

Comment:
Shouldn't this be called `cancel(dev)` or similar?

Reply:
I'm using the same pattern as our `boot` event